### ISSUE TITLE
Very good but difficult to package for distributions

### ISSUE THREAD

**Issue by miraks31** [sijk/qt5-sqlcipher#7]
Hi,

This sqlcipher plugin for Qt5 is very interesting.
But, because we need the full source code of Qt5 (not only headers), this is difficult to use it in a software that must be packaged in linux distribution.

I opened the following incident on Qt to include it in Qt:
https://bugreports.qt.io/browse/QTBUG-46150

As you can see, Qt team advices me to contact you.

What do you think? Are you ready to contribute to Qt with this really interesting plugin?

**Comment by lschuetze** [comment]
He has been very quiet since long time. Maybe you contact him at his email since he does not seem to look at GitHub: simon.knopp@pg.canterbury.ac.nz. Given his LinkedIn profile he still works there.

**Comment by sijk** [comment]
@lschuetze — Yes, I recently finished up my PhD so I haven't had much time for side projects for a while. 

@miraks31 — I agree, this is inconvenient to build. It basically steals the source code for Qt's SQLite driver and links it against SQLCipher instead. Submitting it upstream to Qt is a good suggestion; I'll look into it. I'll have to ask the Qt maintainers whether they want this, and how complex it is to integrate an optional external dependency into their build system. Alternatively, I might look into extracting the SQLite driver sources from Qt so this can be compiled stand-alone as a 3rd-party plugin (provided the licence allows that). I think that might be easier for everyone. 

I haven't touched this code for a year or so, so I'll also have to check whether it builds against the latest versions of Qt and SQLCipher.

**Comment by miraks31** [comment]
Hi Sijk,

Thank you for your answer.
In the bug tracker of Qt, you can find links to contribute. https://bugreports.qt.io/browse/QTBUG-46150
You must know that the SQLITE driver can be build in 2 different modes:
1-The sqlite library is included in the driver. This is not the solution preferred by linux distribution.
2-The system sqlite library is used by the driver. This is the preferred solution.

Could be good to have the same thing with the sqlcipher library.

For information, I am the main developer of an application named Skrooge and will be very happy to use this new plugin when available in Qt.

Thank you again.
Regards.

**Comment by miraks31** [comment]
Hi Sijk,

Did you progress on this topic?

Thank you again.
Regards.

**Comment by sijk** [comment]
Yes, I have made progress on this. I originally tried to integrate it directly into Qt so that it could be merged upstream, but it was a bit too much hassle that way – the linker could get confused when using some combinations of build options because both SQLite and SQLcipher provide the same symbols. 

Instead, I decided to build it as a stand-alone plugin. This is perhaps slightly less convenient for people wanting to use it since it won't be built into Qt, but it's still pretty easy to use. Just build the plugin and drop it next to your application. 

I'll push the new version after work today.
